### PR TITLE
divide: Optimize complexmultiplyconjugate

### PR DESCRIPTION
**Pull request by jdemel**
This commit should give some performance improvement for kernels that
use `_mm256_complexconjugatemul_ps`.

It might be worthwhile to test this PR on Intel machines. So far, I've only tested it on AMD Ryzen.

### CODE REVIEW

File: include/volk/volk_avx_intrinsics.h [hunk 1]
```diff
@@ -47,10 +48,17 @@ static inline __m256 _mm256_conjugate_ps(__m256 x)
     return _mm256_xor_ps(x, conjugator); // conjugate y
 }
 
-static inline __m256 _mm256_complexconjugatemul_ps(__m256 x, __m256 y)
+static inline __m256 _mm256_complexconjugatemul_ps(const __m256 x, const __m256 y)
```
Comment (rear1019) [inline include/volk/volk_avx_intrinsics.h:51]: LGTM

File: kernels/volk/volk_32fc_x2_divide_32fc.h [hunk 2]
```diff
@@ -287,48 +287,58 @@ static inline void volk_32fc_x2_divide_32fc_a_avx(lv_32fc_t* cVector,
...
-    for (; number < num_points; number++) {
-        *c++ = (*a++) / (*b++);
+        complex_result = _mm256_hadd_ps(_mm256_mul_ps(denum01, denum01),
+                                        _mm256_mul_ps(denum23, denum23));
+
+        denum01 = _mm256_shuffle_ps(complex_result, complex_result, 0x50);
+        denum23 = _mm256_shuffle_ps(complex_result, complex_result, 0xfa);
+
+        result0 = _mm256_div_ps(num01, denum01);
+        result1 = _mm256_div_ps(num23, denum23);
```
Comment (rear1019) [inline kernels/volk/volk_32fc_x2_divide_32fc.h:329]: LGTM.

**Review by rear1019 — APPROVED** [review]
The three affected functions are indeed faster on Ryzen (Zen architecture).

On Intel (Ivy Bridge) the aligned versions of `multiply_conjugate` and `multiply_conjugate_add` are faster. Apart from these cases the new code seems to be just as fast as the old.

**Review by michaelld — APPROVED** [review]
LGTM

**Comment by michaelld** [comment]
Relevant CI passes. Maintainer approval. Merging.

**Pull request merged**